modules/core/src/utils/flatten.ts

```diff
@@ -23,14 +23,12 @@
  * or a single value into an array with one value
  * @example flatten([[1, [2]], [3], 4]) => [1, 2, 3, 4]
  * @example flatten(1) => [1]
- * @param {Array} array The array to flatten.
- * @param {Function} filter= - Optional predicate called on each `value` to
+ * @param array The array to flatten.
+ * @param filter= - Optional predicate called on each `value` to
  *   determine if it should be included (pushed onto) the resulting array.
- * @param {Function} map= - Optional transform applied to each array elements.
- * @param {Array} result=[] - Optional array to push value into
- * @return {Array} Returns the new flattened array (new array or `result` if provided)
+ * @return Returns the new flattened array (new array or `result` if provided)
  */
-export function flatten(array, filter = () => true) {
+export function flatten(array: any[], filter = (...args) => true): any[] {
```

Comment:
Can you remove the rest arguments, it has a perf impact.

Reply:
>Can you remove the rest arguments, it has a perf impact.
Done